Assets/Scripts/Tools/EdgeTool.cs

```diff
@@ -34,11 +34,49 @@ public void OnLeftClick(Transform cameraTransform, bool isHit, RaycastHit raycas
       if (!isHit || raycastHit.collider.gameObject.tag != Constants.PhysicalNodeTag)
         return;
 
-      OnActionPerformed(isHit, raycastHit);
+      if (!isHit)
+        return;
+
+      var gameObjectHit = raycastHit.collider.gameObject;
+      var currentlyHitNode = _graphService.FindNodeByGameObject(gameObjectHit);
+
+      // If not a node, don't do anything
+      if (currentlyHitNode == null)
+        return;
+
+      if (_previouslyHitNode == null)
+      {
+        _previouslyHitNode = currentlyHitNode;
+        EnablePreviouslyHitNode();
+        return;
+      }
+
+      _graphService.AddEdge(currentlyHitNode, _previouslyHitNode);
+      DisablePreviouslyHitNode();
+    }
+
+    private void EnablePreviouslyHitNode()
+    {
+      _previouslyHitNode.gameObject.EnableGlow();
+      var glowColor = _createGlowColor;
+      _previouslyHitNode.gameObject.SetGlow(glowColor * _nodeHitGlowStrength);
+    }
+
+    private void DisablePreviouslyHitNode()
+    {
+      _previouslyHitNode.gameObject.DisableGlow();
+      _previouslyHitNode = null;
     }
 
     public void OnRightClick(Transform cameraTransform, bool isHit, RaycastHit raycastHit)
     {
+
+      if (isHit && raycastHit.collider.gameObject.tag == Constants.PhysicalNodeTag && _previouslyHitNode != null)
```

Comment:
This logic is complicated enough that I would like it to be extracted to a function (or commented)